src/plugins/wizard/public/application/components/top_nav.tsx

```diff
@@ -30,6 +30,7 @@ export const TopNav = () => {
   );
 
   const savedWizardVis = useSavedWizardVis(visualizationIdFromUrl);
+  const indexPattern = useIndexPatterns().selected;
```

Comment:
We still pass index pattern in `visualizationState`. You can actually remove that from `visualizationState` and use it here. It is just the index id, but looking at the [extract references](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.2/src/plugins/data/common/search/search_source/extract_references.ts#L41) function, it should be okay to just pass the id

Reply:
I will remove the index pattern id from the visualizatonState so i will not get saved there again. 

However, for the indexPattern type in the searchSourceFields, i can not only pass in the id of the index pattern(they make it strict so i have to include other info for the index pattern), so i think its easier here to just pass in an entire index pattern project.